### PR TITLE
b/188654656:add jwt_pad_forward_payload_header

### PR DESCRIPTION
**Pull request by TAOXUY**
GoogleCloudPlatform/esp-v2#178

### CODE REVIEW

File: tests/integration_test/jwt_auth_integration_test/jwt_auth_integration_test.go [hunk 1]
```diff
@@ -441,22 +396,89 @@ func TestFrontendAndBackendAuthHeaders(t *testing.T) {
...
+
+			if err != nil {
+				t.Fatalf("Test Desc(%s): %v", tc.desc, err)
+			}
+
+			gotResp := string(resp)
+			for wantKey, wantValue := range tc.wantHeaders {
+				wantHeader := fmt.Sprintf(`"%v": "%v"`, wantKey, wantValue)
+				if !strings.Contains(gotResp, wantHeader) {
+					t.Fatalf("Test Desc(%s) failed,\n  got: %v\n want: %v", tc.desc, gotResp, wantHeader)
```
Comment (shuoyang2016) [inline tests/integration_test/jwt_auth_integration_test/jwt_auth_integration_test.go:479]: nit: This is not precise. Maybe "got response: %v. Want header %v in the response."

Comment (TAOXUY) [reply tests/integration_test/jwt_auth_integration_test/jwt_auth_integration_test.go]: Done.

**Review by shuoyang2016 — APPROVED** [review]

File: tests/env/testdata/fake_jwt.go [hunk 2]
```diff
@@ -296,6 +296,22 @@ const (
...
+		"RmJwb0lVcXJZOHQyenBBMnFYZkNtcjVWTzVaRXI0UnpIVV8tZW52dlEiLCJ0eXAiOiJK" +
+		"V1QifQ.eyJhdWQiOiJuZWVkLXBhZCIsImV4cCI6MzMxNjY1MzYxMjQsImlhdCI6MTYzM" +
+		"DUzNjEyNCwiaXNzIjoiYXBpLXByb3h5LXRlc3RpbmdAY2xvdWQuZ29vZyIsInN1YiI6I" +
+		"mFwaS1wcm94eS10ZXN0aW5nQGNsb3VkLmdvb2cifQ.egwbVVrwywLJXBLCg00mg-hS7d" +
+		"qsdl2frZdG4rDd_6Vqpt_cTlZZAAC3bppSMCWpDn96y-2-K0zxzjkl3JzcZ8o5KY1F5p" +
+		"oVwuOMsxr7WJatrj1-96NSCE4TXuyqtjG2p_0l3vhbE87bA9TN-XkK9-0NOMxUDWZOMK" +
+		"n_SM18XT4s2XFZa9z8fFMf2fkmoPiwu9sVKLYwhv94xPwU4VWJs-AWpkyUBIrwR21n8H" +
+		"WEIDqb43oK6RzngfkJV4qj253W2KGCkpoq91C06FWptAMwFIyR6Dnu8BAlXyl4le9gxd" +
+		"ysPObKgkjxcyyKawhuzZowWy2Kk1fcAUXzuS6tlUkVzQ"
+	FakeCloudTokenSingleAudienc3Payload = "eyJhdWQiOiJuZWVkLXBhZCIsImV4cCI6" +
```
Comment (nareddyt) [inline tests/env/testdata/fake_jwt.go:311]: nit: Audience

**Review by nareddyt — APPROVED** [review]

**Pull request merged**